pkg/apply/applier_test.go

```diff
@@ -364,6 +441,80 @@ func TestValidateNamespace(t *testing.T) {
 	}
 }
 
+func TestReadAndPrepareObjects(t *testing.T) {
+	testCases := map[string]struct {
+		resources     []*resource.Info
+		expectedError bool
+	}{
+		"no grouping object": {
+			resources:     []*resource.Info{obj1Info},
+			expectedError: true,
+		},
+		"multiple grouping objects": {
+			resources:     []*resource.Info{groupingObjInfo, groupingObjInfo},
+			expectedError: true,
+		},
+		"only grouping object": {
```

Comment:
Just to be consistent, should we put the `expectedError: false` here (instead of being implied)?

Reply:
Agree